src/User_Command.php

```diff
@@ -280,8 +280,10 @@ public function delete( $args, $assoc_args ) {
 			WP_CLI::error( 'Reassigning content to a different user is not supported on multisite.' );
 		}
 
-		if ( ! $reassign ) {
-			WP_CLI::confirm( '--reassign parameter not passed. All associated posts will be deleted. Proceed?', $assoc_args );
+		$is_reassign_valid = ( false === get_userdata( $reassign ) ) ? false : true;
+
+		if ( ! $reassign || ! $is_reassign_valid ) {
+			WP_CLI::confirm( '--reassign parameter not passed or invalid. All associated posts will be deleted. Proceed?', $assoc_args );
```

Comment:
I think we should actually have two separate confirmation prompts, so we can use more precise messages:

```php
if ( ! $is_reassign_valid ) {
	WP_CLI::confirm( '--reassign parameter is invalid. All associated posts will be deleted. Proceed?', $assoc_args );
}

if ( ! $reassign ) {
	WP_CLI::confirm( '--reassign parameter not passed. All associated posts will be deleted. Proceed?', $assoc_args );
}
```

Here's an example where session input is passed to the command:

https://github.com/wp-cli/entity-command/blob/fd187722baecc28b955b0a42d0b780042ca6e896/features/post-meta-clean-duplicates.feature#L33

You can follow a similar approach for writing a test here.

Reply:
PR has been updated.